x/staking/migrations/v5/store.go

```diff
@@ -0,0 +1,68 @@
+package v5
+
+import (
+	"sort"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// MigrateStore performs in-place store migrations from v3 to v4.
+// this migrateStore will remove the ubdEntries with same creation_height
+// and create a new ubdEntry with updated balance and initial_balance
+func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec) error {
+	store := ctx.KVStore(storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.UnbondingDelegationKey)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		ubd := types.MustUnmarshalUBD(cdc, iterator.Value())
+
+		entriesAtSameCreationHeight := make(map[int64][]types.UnbondingDelegationEntry)
+
+		var creationHeights []int64
+
+		for _, ubdEntry := range ubd.Entries {
+			entriesAtSameCreationHeight[ubdEntry.CreationHeight] = append(entriesAtSameCreationHeight[ubdEntry.CreationHeight], ubdEntry)
+			creationHeights = append(creationHeights, ubdEntry.CreationHeight)
+		}
+
+		// clear the old ubdEntries
+		ubd.Entries = nil
+
+		sort.Slice(creationHeights, func(i, j int) bool { return creationHeights[i] < creationHeights[j] })
+
+		for index := 0; index < len(creationHeights); index++ {
```

Comment:
why not just range over `creationHeights`? i.e. `for _, h := range creationHeights {}`